### PR TITLE
CI config updates

### PR DESCRIPTION
**Pull request by garrettwrong**
Our CI machines are configured a little funny.

This patch provides three changes.

- Creates and cleans up a tmp dir
- Writes and uses a custom ASPIRE configuration
- Migrates some temporal storage from `/tmp` towards `/var/ci/...`

### CODE REVIEW

**Comment by garrettwrong** [comment]
Cool, using Github's temp dir is much cleaner. I realized because the self hosted runner is only running a single job at a time, we don't need to worry about unique temp dirs.  I'll leave the prior commit in the history in case we need it later though.

**Comment by garrettwrong** [comment]
Here is a link to one of the relevant doc pages.

https://docs.github.com/en/actions/learn-github-actions/environment-variables

**Comment by garrettwrong** [comment]
Unfortunately, looks like we can't easily use the GH temp dir, because it results in a path that has too many characters for a socket filename.  Honestly didn't see that one coming 😇 . My new thing I learned today...

**Review by chris-langfield — APPROVED** [review]
Looks good! Logging everything out is a good idea

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -83,5 +83,24 @@ jobs:
...
+        echo "WORK_DIR=${WORK_DIR}" >> $GITHUB_ENV
+        echo -e "ray:\n    temp_dir: ${WORK_DIR}\n" > ${WORK_DIR}/config.yaml
+        echo "Log the config: ${WORK_DIR}/config.yaml"
+        cat ${WORK_DIR}/config.yaml
     - name: Run
-      run: pytest
+      run: |
+        ASPIREDIR=${{ env.WORK_DIR }} python -c \
+        "import aspire; print(aspire.config['ray']['temp_dir'])"
+        ASPIREDIR=${{ env.WORK_DIR }} pytest
```
Comment (j-c-c) [inline .github/workflows/workflow.yml:104]: Should we randomize pytest here?

**Review by j-c-c — COMMENTED** [review]
Looks good! Just one question.

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (garrettwrong) [reply .github/workflows/workflow.yml:104]: why

Comment (j-c-c) [reply .github/workflows/workflow.yml:104]: Wasn't sure if that was a practice we wanted to continue after resolving the azure pipelines timeouts. If not, then disregard.

**Review by j-c-c — APPROVED** [review]
🚀

**Comment by garrettwrong** [comment]
Merging

**Pull request merged**